erpnext/buying/doctype/supplier/supplier.py

```diff
@@ -62,6 +62,13 @@ def validate(self):
 
 		validate_party_accounts(self)
 		self.validate_internal_supplier()
+		self.validate_portal_users()
+
+	def validate_portal_users(self):
+		for portal_user in self.portal_users:
+			user_roles = frappe.get_roles(portal_user.user)
+			if "Supplier" not in user_roles:
+				frappe.throw(_("Portal Users must have Supplier Role."))
```

Comment:
Let's automatically add this role. This role doesn't grant any permission besides considering user as "Supplier". 

- Use ORM, so if user has write to add this role then only they'll be able to do it. 
- We should give alert with `frappe.msgprint(alert=True` that we have automatically added this role to user.